Assets/Scripts/Systems/Modules/ModulesInfo.cs

```diff
@@ -7,97 +7,169 @@
 
 namespace Systems.Modules
 {
+    /// <summary>
+    ///     Stores runtime information on Modules and Module Grid.
+    ///     Saves and Loads Modules and Module Grid.
+    ///     Will init with default values from ShipData if there is no save data.
+    /// </summary>
     [Serializable]
     public class ModulesInfo : MonoBehaviour, ISavable
     {
-        public Module[,] Grid;
-        public List<Module> Modules;
-        public int RowHeight;
-        public int ColumnLength;
+        private Module[,] _grid;
+        private ModuleSlot[,] _slotGrid;
+        private List<Module> _modules;
+        private List<ModuleSlot> _moduleSlots;
+        private int _rowHeight;
+        private int _columnLength;
 
         private ShipInfo _info;
 
+        public string id => "module_grid";
+        public int RowHeight => _rowHeight;
+        public int ColumnLength => _columnLength;
+        public List<Module> Modules => _modules;
+
         private void Awake()
         {
             _info = GetComponent<ShipInfo>();
-            if (Grid == null)
+            if (_grid == null)
             {
-                Grid = ModuleList.ListsToGrid(_info.Data.ModuleGrid);
+                _rowHeight = _info.Data.ModuleGridHeight;
+                _columnLength = _info.Data.ModuleGridWidth;
+                _grid = new Module[_rowHeight, _columnLength];
+                _modules = _info.Data.ModuleList;
+                _moduleSlots = _info.Data.ModuleSlots;
+                foreach (Module module in _modules)
+                {
+                    AddModule(module);
+                }
+
+                foreach (ModuleSlot moduleSlot in _moduleSlots)
+                {
+                    AddModuleSlot(moduleSlot);
+                }
             }
+        }
```

Comment:
This should probably be Start so that ship info has been initialized from save data before you access it